blocktx/store/interface.go

```diff
@@ -28,5 +27,10 @@ type Interface interface {
 	MarkBlockAsDone(ctx context.Context, hash *chainhash.Hash, size uint64, txCount uint64) error
 	GetBlockGaps(ctx context.Context, heightRange int) ([]*BlockGap, error)
 	ClearBlocktxTable(ctx context.Context, retentionDays int32, table string) (*blocktx_api.ClearDataResponse, error)
+
+	SetBlockProcessing(ctx context.Context, hash *chainhash.Hash, processedBy string) (string, error)
+	GetBlockHashesProcessingInProgress(ctx context.Context, processedBy string) ([]*chainhash.Hash, error)
```

Comment:
How can we be processing many blocks at the same time? Isn't that dangerous? Shouldn't we be processing a new block if we have prev_block_hash already processed?

Reply:
If a block announcement comes in [here](https://github.com/bitcoin-sv/arc/blob/ec2b718b940d268fdfc68464bbf05a23faf042f0/blocktx/peer_handler.go#L426) then only one `blocktx` instance can process it, because only the first instance which tries to set the block processing [here](https://github.com/bitcoin-sv/arc/blob/ec2b718b940d268fdfc68464bbf05a23faf042f0/blocktx/peer_handler.go#L218) succeeds. Every other `blocktx` instance will fail here with error `store.ErrBlockProcessingDuplicateKey`. That ensures that any `blocktx` can process a block, but only one `blocktx` will process a certain block.

There is no danger in processing a block if the previous one is not processed yet. There is the `FillGaps` [go routine](https://github.com/bitcoin-sv/arc/blob/ec2b718b940d268fdfc68464bbf05a23faf042f0/blocktx/peer_handler.go#L203) which ensures that all blocks are eventually getting processed